alicloud/service_alicloud_ecs.go

```diff
@@ -463,3 +463,26 @@ func (client *AliyunClient) QueryInstancesWithKeyPair(region common.Region, inst
 	}
 	return instance_ids, instanceList, nil
 }
+
+func GetAllRouterInterfaceSpec() (specifications []string) {
+	specifications = append(specifications, string(ecs.Large1), string(ecs.Large2),
+		string(ecs.Small1), string(ecs.Small2), string(ecs.Small5), string(ecs.Middle1),
+		string(ecs.Middle2), string(ecs.Middle5), string(Negative))
+	return
+}
+
+func (client *AliyunClient) JudgeRegionValidation(key string, region common.Region) error {
+	regions, err := client.ecsconn.DescribeRegions()
+	if err != nil {
+		return fmt.Errorf("DescribeRegions got an error: %#v", err)
+	}
+
+	var rs []common.Region
+	for _, v := range regions {
+		if v.RegionId == region {
+			return nil
+		}
+		rs = append(rs, v.RegionId)
+	}
+	return fmt.Errorf("The '%s' must be one of %v.", key, rs)
```

Comment:
Please modify fmt.Errorf("The '%s' must be one of %v.", key, rs) to fmt.Errorf("'%s' is invalid. Expected on %.", key, strings.Join(rs, ", ")).